ui/page/root/trade_page.go

```diff
@@ -86,16 +82,15 @@ func (pg *TradePage) OnNavigatedTo() {
 // displayed.
 // Part of the load.Page interface.
 func (pg *TradePage) HandleUserInteractions() {
-	if activeTab := pg.CurrentPage(); activeTab != nil {
-		activeTab.HandleUserInteractions()
+	if pg.tab.Changed() {
+		if pg.tab.SelectedIndex() == 1 {
+			pg.Display(exchange.NewCreateOrderPage(pg.Load))
+		} else if pg.CurrentPageID() != dcrdex.DCRDEXPageID {
+			pg.Display(dcrdex.NewDEXPage(pg.Load))
+		}
 	}
 
-	if pg.tab.SelectedIndex() == 0 {
-		pg.Display(dcrdex.NewDEXPage(pg.Load))
-	}
-	if pg.tab.SelectedIndex() == 1 {
-		pg.Display(exchange.NewCreateOrderPage(pg.Load))
-	}
+	pg.CurrentPage().HandleUserInteractions()
```

Comment:
Forwarding User Interaction should happen `after` a new page has been assigned if we hit the tab changed if branch. Same thing applies to other master pages.